### PR TITLE
Lint fixes

### PR DESCRIPTION
**Pull request by igorRoog**
**Merge to ui-revamp-feature branch!**

### CODE REVIEW

File: src/api/types.ts [hunk 1]
```diff
@@ -21,13 +21,13 @@ interface ScorecardCreator {
 
 export interface Scorecard {
   creator: ScorecardCreator;
+  description?: string;
+  filter?: EntityFilter | null;
```
Comment (FoodProduct) [inline src/api/types.ts:25]: In what scenario does the `filter` property not exist on a Scorecard?

File: src/api/types.ts [hunk 2]
```diff
@@ -21,13 +21,13 @@ interface ScorecardCreator {
 
 export interface Scorecard {
   creator: ScorecardCreator;
+  description?: string;
```
Comment (FoodProduct) [inline src/api/types.ts:24]: Should this be 
```suggestion
  description: string | null;
```
?

File: src/components/Homepage/HomepageInsightCard.tsx [hunk 3]
```diff
@@ -56,7 +56,8 @@ export const HomepageInsightCard = ({
         currEntity => currEntity.id === actionItemInsight.entityId,
       );
       const initiative = initiatives?.find(
-        currInitiative => currInitiative.id === actionItemInsight.initiativeId,
+        currInitiative =>
+          currInitiative.id === actionItemInsight.initiativeId.toString(),
```
Comment (FoodProduct) [inline src/components/Homepage/HomepageInsightCard.tsx:60]: 😿 that our mistyping is spilling over to the backstage plugin

File: src/api/types.ts (hunk 1 above)
Comment (igorRoog) [reply src/api/types.ts:25]: I don't know anything about such scenario, however I had to add `?` as in `hooks.ts` is `useScorecardCompareFn` defined, which uses `customCompareFn` (which may be defined by user (?)) which (for some reason) calls `nullsToUndefined` on each scorecard first.

It was added there for some reason, so I didn't want to change that, however I had to extend Scorecard type definition.

File: src/api/types.ts (hunk 2 above)
Comment (igorRoog) [reply src/api/types.ts:24]: I'll lead to o lot of cascade changes in multiple files/types so I would leave it as it is.

File: src/api/types.ts (hunk 1 above)
Comment (FoodProduct) [reply src/api/types.ts:25]: Ugh alright

**Pull request merged**